src/main/java/service/impl/DataConverterImpl.java

```diff
@@ -0,0 +1,48 @@
+package service.impl;
+
+import exception.InvalidDataException;
+import java.util.ArrayList;
+import java.util.List;
+import model.FruitTransaction;
+import model.OperationType;
+import service.DataConverter;
+
+public class DataConverterImpl implements DataConverter {
+    private static final int OPERATION_INDEX = 0;
+    private static final int FRUIT_NAME_INDEX = 1;
+    private static final int QUANTITY_INDEX = 2;
+    private static final String SEPARATOR = ",";
+
+    @Override
+    public List<FruitTransaction> fruitList(List<String> inputInfo) {
+        List<FruitTransaction> fruitTransactionList = new ArrayList<>();
+        for (String line : inputInfo) {
+            String[] splitLine = line.split(SEPARATOR);
+            if (splitLine.length != 3) {
+                throw new InvalidDataException("Invalid input data format: " + line);
+            }
+            String operation = splitLine[OPERATION_INDEX].trim();
+            String fruitName = splitLine[FRUIT_NAME_INDEX].trim();
+            int quantity = Integer.parseInt(splitLine[QUANTITY_INDEX].trim());
+            if (operation.equals((OperationType.BALANCE.getName()))) {

```

Comment:
```suggestion
            if (operation.equals(OperationType.BALANCE.getName())) {
```
it's not appropriate logic for this service to move to BalanceOperationService. And do you really need it? If you want just update balance you can’t  do it